Add utils to replace torchtune SDPA with ET Custom SDPA

tarun292 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #7532
* __->__ #7531
* #7530

Differential Revision: [D67878161](https://our.internmc.facebook.com/intern/diff/D67878161)

tarun292 committed:
Add utils to replace torchtune SDPA with ET Custom SDPA

[ghstack-poisoned]

tarun292 commented:
@tarun292 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67878161).

dvorjackz commented on extension/llm/modules/attention.py:
```diff
@@ -310,7 +311,9 @@ def false_fn(y):
             self.kv_cache.cache_pos.copy_(cache_pos)
 
-        output = self._sdpa(q, k, v, b, s_x, mask=mask)
+        if input_pos is None:
```
This will break the non kv cache version of the text decoder, in that case `input_pos` is also None

tarun292 committed:
Update on "Add utils to replace torchtune SDPA with ET Custom SDPA"


Differential Revision: [D67878161](https://our.internmc.facebook.com/intern/diff/D67878161)

[ghstack-poisoned]

tarun292 commented:
@tarun292 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67878161).

tarun292 committed:
Update on "Add utils to replace torchtune SDPA with ET Custom SDPA"


Differential Revision: [D67878161](https://our.internmc.facebook.com/intern/diff/D67878161)

[ghstack-poisoned]

tarun292 commented:
@tarun292 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D67878161).
